src/toast/Toast.tsx

```diff
@@ -0,0 +1,163 @@
+    SUCCESS: "success-standard",
+    DANGER: "error-standard",
+    INFO: "info-standard",
+};
+
+interface ToastProps {
+    displayToast: (fn: Function) => void;
+}
+
+type ToastState = Readonly<typeof initialState>;
+
+export default class Toast extends React.Component<ToastProps> {
+    readonly state: ToastState = initialState;
+
+    constructor(props: ToastProps) {
+        super(props);
+        props.displayToast(this.show_toast);
+    }
+
+    /* Show the toaster unless another toaster is already in display */
+    show_toast = (message: string, options: ToastOptions) => {
+        //Default display time for toast is 3 sec
+        //Default type is SUCCESS
+        //Default position is top-right
+        const hideTime = (options.hideAfter === undefined ? DEFAULT_HIDE_TIME : options.hideAfter) * MILLISEC;
+        const toastType = options.type === undefined ? MessageType.SUCCESS : options.type;
+        const pos = options.position === undefined ? Direction.TOP_RIGHT : options.position;
+
+        //As this is a simple toast, allowing one toast to be shown at a time
+        if (this.state.isToastRendering === false) {
+            //set the state to show the toast and setTimeout to hide the toast
+            this.setState(
+                {
+                    isToastRendering: true,
+                    msg: message,
+                    type: toastType,
+                    position: pos,
+                    hideAfter: hideTime,
+                },
+                () => {
+                    setTimeout(
+                        () =>
+                            this.setState({
+                                isToastRendering: false,
+                                msg: EMPTY_MESSAGE,
+                                type: MessageType.SUCCESS,
+                                position: Direction.TOP_RIGHT,
+                                hideAfter: DEFAULT_HIDE_TIME,
+                            }),
+                        hideTime,
+                    );
+                },
+            );
+        }
+    };
+
+    render() {
+        let classes = "";
+        let classname = "";
+        if (this.state.isToastRendering === true) {
+            //create styles based on the position given
+            classes = TOAST_CLASSES.VISIBLE + " ";
+            classname = (this.state.position || Direction.TOP_RIGHT).includes(Direction.BOTTOM)
+                ? TOAST_CLASSES.BOTTOM + " " + TOAST_CLASSES.MARGIN_BOTTOM
+                : TOAST_CLASSES.TOP + " " + TOAST_CLASSES.MARGIN_TOP;
+            classes = classes + " " + classname + " ";
+            classname = (this.state.position || Direction.TOP_RIGHT).includes(Direction.RIGHT)
+                ? TOAST_CLASSES.RIGHT + " " + TOAST_CLASSES.MARGIN_RIGHT
+                : TOAST_CLASSES.LEFT + " " + TOAST_CLASSES.MARGIN_LEFT;
+            classes = classes + " " + classname;
+        } else {
+            classes = TOAST_CLASSES.INVISIBLE + " ";
```

Comment:
We should have to style the div as hidden/invisible.  We can just check the state and either render the div into the DOM or completely omit it.  Please check Model for some pointers on how to do this.

Reply:
Updated code.